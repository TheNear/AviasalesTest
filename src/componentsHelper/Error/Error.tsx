import React from "react";
import commonStyle from "../../assets/css/common.module.css";
import style from "./Error.module.css";

interface IError {
  refreshTicketsList: () => void;
}

const Error: React.FC<IError> = ({ refreshTicketsList }) => {
  return (
    <div className={`${commonStyle.box} ${style.wrapper}`}>
      <p className={style.text}>Ошибка загрузки билетов!</p>
      <button onClick={refreshTicketsList} className={style.button} type="button">
        Попробовать еще раз?
      </button>
    </div>
  );
};

export { Error };
