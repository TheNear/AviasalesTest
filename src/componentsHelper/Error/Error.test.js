import React from "react";
import { shallow } from "enzyme";
import { Error } from "./Error";

describe("Test custom Error component", () => {
  it("Render empty Error", () => {
    const wrapper = shallow(<Error />);
    expect(wrapper).toMatchSnapshot();
  });
});
