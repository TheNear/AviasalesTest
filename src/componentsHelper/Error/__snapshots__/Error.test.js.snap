// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test custom Error component Render empty Error 1`] = `
<div
  className="box wrapper"
>
  <p
    className="text"
  >
    Ошибка загрузки билетов!
  </p>
  <button
    className="button"
    type="button"
  >
    Попробовать еще раз?
  </button>
</div>
`;
